### PR TITLE
PXP-596: Optimize the logs rendering on tui

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary
Update the various logics on logs to make sure that wukong TUI to improve the performance on lower end devices

Ticket: https://mindvalley.atlassian.net/browse/PXP-596

## What's Changed

- [x] Change the logic of storing the logs to make it less heavy during re-rendering
- [x] Limit the log history to 1_000
- [x] Reduce the tick rate
- [x] Set cursor to fixed place for now to avoid screen scrolling on new printLn

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/145#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — DISMISSED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/145#" title="Author self-approved">mfauzaan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/145#pullrequestreview-1623673008" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cli/src/commands/tui/app.rs [hunk 1]
```diff
@@ -104,9 +105,10 @@ impl App {
                 builds: vec![],
                 deployments: vec![],
                 last_log_entry_timestamp: None,
-                log_entries_hash_map: HashMap::new(),
-                log_entries_ids: vec![],
                 has_log_errors: false,
+                log_entries_length: 0,
```
Comment (jk-gan) [inline cli/src/commands/tui/app.rs:109]: I believe we don't need this, we should be able to get the len using `log_entries.len()`

File: cli/src/commands/tui/mod.rs [hunk 2]
```diff
@@ -146,8 +144,18 @@ pub async fn start_ui(app: &Arc<Mutex<App>>) -> std::io::Result<bool> {
...
             break;
         }
+
+        if is_first_render {
+            // fetch data on the first frame
+            app_ref.dispatch(NetworkEvent::FetchDeployments).await;
+            app_ref.dispatch(NetworkEvent::FetchBuilds).await;
+
+            is_first_render = false;
+        }
```
Comment (jk-gan) [inline cli/src/commands/tui/mod.rs:154]: Any reason to move this one?

File: cli/src/commands/tui/app.rs [hunk 3]
```diff
@@ -34,9 +34,10 @@ pub struct State {
     // fetch data
     pub builds: Vec<Build>,
     pub deployments: Vec<Deployment>,
-    pub log_entries_hash_map: HashMap<String, LogEntry>,
     pub has_log_errors: bool,
-    pub log_entries_ids: Vec<String>,
+    pub log_entries: Vec<LogEntry>,
+    pub log_entries_length: usize,
+    pub max_log_entries_length: usize,
```
Comment (jk-gan) [inline cli/src/commands/tui/app.rs:40]: this can be constant

**Review by jk-gan — DISMISSED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/145#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/145#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jk-gan — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/145#pullrequestreview-1628565547" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/145#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**